File: trunk/src/dotnet/yaxgl/YaxglLibrary/Container/RadioGroup.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml;
namespace de.yaxgl
{
    public class RadioGroup : Container,Containable
    {
        
        public RadioGroup(XmlElement rootElement,Container owner, string ID) 
        {
            this.owner = owner;
            this.ID = ID;
            this.control = new System.Windows.Forms.Panel();
            
            /*register events*/

            parseXML(rootElement);
            
        }

        public override void initializeNativeControl(XmlElement xmlElement)
        {
            setBounds(Convert.ToInt32(xmlElement.Attributes["xpos"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["ypos"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["width"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["height"].InnerText));

            check(xmlElement.Attributes["checked"].InnerText);    
        }

        private void check(string radioButtonId)
        {
            if (components.ContainsKey(radioButtonId))
                ((RadioButton)components[radioButtonId]).setChecked(true);
        }

        public RadioButton getCheckedRadioButton()
        {
            RadioButton retVal=null;
            foreach (KeyValuePair<string,Containable> keyValue in components)
            {
                if (((RadioButton)keyValue.Value).isChecked())
                    retVal=(RadioButton)keyValue.Value;  
            }
            return retVal;
        }
        
        public override void notifyEvent(Component control, EventArgs eventArgs)
        {
            owner.notifyEvent(control, eventArgs);
        }

    }
}

File: trunk/src/java/YaxglLibrary/de/yaxgl/Container/RadioGroup.java
package de.yaxgl.Container;

import org.eclipse.swt.SWT;
import org.w3c.dom.Element;

import de.yaxgl.Base.Component;
import de.yaxgl.Base.Containable;
import de.yaxgl.Controls.RadioButton;
import de.yaxgl.EventDispatcher.EventArgs;

@SuppressWarnings("serial")
public class RadioGroup extends Container implements Containable{

	public RadioGroup(Element rootElement,Container owner,String ID)
	{
		this.owner = owner;
		this.ID=ID;
		org.eclipse.swt.widgets.Composite composite=(org.eclipse.swt.widgets.Composite)this.owner.getNativeComponent();                
        this.control = new org.eclipse.swt.widgets.Composite(composite,SWT.NONE); 
        try {
			parseXML(rootElement);
		} catch (Exception e) {
			System.out.println(e.getMessage());
		}
	}
	
	@Override
	public void initializeNativeControl(Element xmlElement) {
		setBounds(Integer.valueOf(xmlElement.getAttribute("xpos")),
				Integer.valueOf(xmlElement.getAttribute("ypos")),
				Integer.valueOf(xmlElement.getAttribute("width")),
				Integer.valueOf(xmlElement.getAttribute("height")));
		
		check(xmlElement.getAttribute("checked"));
	}
	
	@Override
	public void notifyEvent(Component control, EventArgs eventArgs) {
		this.owner.notifyEvent(control, eventArgs);
	}
	
	public void check(String radioButtonId)
    {
		if (components.containsKey(radioButtonId))
            ((RadioButton)components.get(radioButtonId)).setChecked(true);
    }

    public RadioButton getCheckedRadioButton()
    {
        RadioButton retVal=null;
        
        for(Containable radio :  components.values())
        {
            if (((RadioButton)radio).isChecked())
                retVal=(RadioButton)radio;  
        }
        return retVal;
    }
	

}
